src/fuse_reduce.cpp

```diff
@@ -210,6 +210,32 @@ static void create_reduce_modules(module_pass_manager& mpm)
     }
 }
 
+namespace {
+MIGRAPHX_PRED_MATCHER(used_once_except_broadcast, instruction_ref ins)
```

Comment:
add unit-tests for this one

Reply:
I added a test for this case.

Reply:
Where ? I expected test inside `test/matcher.cpp`

Reply:
The test case was added to fuse_reduce.cpp. `test/matcher.cpp` is for testing the `migraphx/matcher.hpp`. This matcher does not belong there, its specific to fuse_reduce which is why its a internal/private function.